temporalcli/commands.schedule_test.go

```diff
@@ -0,0 +1,361 @@
+package temporalcli_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"regexp"
+	"time"
+
+	"github.com/temporalio/cli/temporalcli"
+	"go.temporal.io/sdk/workflow"
+)
+
+func (s *SharedServerSuite) schedId() string   { return s.testVar("sched") }
+func (s *SharedServerSuite) schedWfId() string { return s.testVar("my-wf-id") }
+
+func (s *SharedServerSuite) createSchedule(args ...string) *CommandResult {
+	s.Worker.OnDevWorkflow(func(ctx workflow.Context, input any) (any, error) {
+		return nil, workflow.Sleep(ctx, 10*time.Second)
+	})
+	s.T().Cleanup(func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+		options := temporalcli.CommandOptions{
+			Stdout: io.Discard,
+			Stderr: io.Discard,
+			Args: []string{
+				"schedule", "delete",
+				"--address", s.Address(),
+				"-s", s.schedId(),
+			},
+			Fail: func(error) {},
+		}
+		temporalcli.Execute(ctx, options)
+	})
+	return s.Execute(append([]string{
+		"schedule", "create",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+		"--task-queue", s.Worker.Options.TaskQueue,
+		"--type", "DevWorkflow",
+		"--workflow-id", s.schedWfId(),
+	}, args...)...,
+	)
+}
+
+func (s *SharedServerSuite) TestSchedule_Create() {
+	res := s.createSchedule("--interval", "10d")
+	s.NoError(res.Err)
+}
+
+func (s *SharedServerSuite) TestSchedule_Delete() {
+	res := s.createSchedule("--interval", "10d")
+	s.NoError(res.Err)
+
+	// check exists
+	res = s.Execute(
+		"schedule", "describe",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+	)
+	s.NoError(res.Err)
+
+	res = s.Execute(
+		"schedule", "delete",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+	)
+	s.NoError(res.Err)
+
+	// doesn't exist anymore
+	res = s.Execute(
+		"schedule", "describe",
+		"--address", s.Address(),
+		"-s", s.schedId(),
+	)
+	s.Error(res.Err)
+}
+
+func (s *SharedServerSuite) TestSchedule_Describe() {
+	res := s.createSchedule("--interval", "2s")
+	s.NoError(res.Err)
+
+	time.Sleep(4 * time.Second) // run at least once
```

Comment:
Can we avoid sleeps in tests? Can we do some kind of manual trigger/backfill here?